sdk/core/azure-core/src/main/java/com/azure/core/cryptography/AsyncKeyEncryptionKeyResolver.java

```diff
@@ -16,5 +16,5 @@
      * @param keyId The key identifier of the key encryption key to retrieve
      * @return The key encryption key corresponding to the specified {@code keyId}
      */
-    Mono<AsyncKeyEncryptionKey> resolveKey(String keyId);
+    Mono<? extends AsyncKeyEncryptionKey> buildAsyncKeyEncryptionKey(String keyId);
```

Comment:
Is `build` the right prefix here? Builder pattern also uses `build` prefix. Would be better to use a different prefix like `create` or `generate` or something similar.

Reply:
The thinking is that this will be used primarily within builders, as it is for the Key Vault CryptographyClientBuilder, and therefore it will feel at home.